### PR TITLE
Fetch Proposal metadata - Cosmos v1

### PR DESCRIPTION
**Pull request by mhagel**
## Link to Issue
Closes: #4184

## Description of Changes
- Adds useProposalMetadata hook for Cosmos V1
- lazy-loads metadata if v1 proposal.metadata IRI is present
- updates proposal in store after lazy-load
- uses IPFS proxy if ipfs file. Non-IPFS resources are disabled for now.

## Test Plan
- Added tests for proposals.spec E2e
- CA (click around) tested on local:
  - http://localhost:8080/kyve/proposals -> expect titles populated for Props 5 and 6
  - click to Prop 5 or 6 -> expect title and description populated
  - click to another community, paste and go directly to http://localhost:8080/kyve/proposal/6 -> expect title and description to load

## Other Considerations

- Follow-up to enable non-IPFS: https://github.com/hicommonwealth/commonwealth/issues/4233

### CODE REVIEW

File: packages/commonwealth/client/scripts/hooks/cosmos/useProposalMetadata.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      proposal?.data?.metadata &&
+      !hasFetchedDataRef.current &&
+      !proposal.data.title // TODO: remove this check if we want to show all metadata.
+      //See https://github.com/hicommonwealth/commonwealth/issues/4187
+    ) {
+      const fileURI = proposal.data.metadata.replace('ipfs://', '');
+      const isIPFSFile = isIPFS.cid(fileURI);
+      const metadataUri = isIPFSFile
+        ? `${app.serverUrl()}/ipfsProxy?hash=${fileURI}`
+        : fileURI;
```
Comment (mhagel) [inline packages/commonwealth/client/scripts/hooks/cosmos/useProposalMetadata.tsx:33]: @jnaviask I am allowing it to fetch any URL put into the proposal.metadata on-chain object. I wonder if this is something we should disable until the need comes? It is easy to see somebody using a regular cloud bucket, or even a centralized database API endpoint, who knows.

I just wonder if it is a good idea to keep this open-ended.

Comment (jnaviask) [reply packages/commonwealth/client/scripts/hooks/cosmos/useProposalMetadata.tsx:33]: @mhagel let's disable arbitrary fetching until we encounter a non-IPFS URL. Too much risk.

Comment (mhagel) [reply packages/commonwealth/client/scripts/hooks/cosmos/useProposalMetadata.tsx]: Sounds good. I made a new [issue](https://github.com/hicommonwealth/commonwealth/issues/4233) and disabled non-IPFS for now.

**Review by jnaviask — APPROVED** [review]
Great work!

**Pull request merged**